Update branch deletion instructions for Vercel (#1597)
neondatabase/website: content/docs/guides/vercel.md

          The **Create a branch for your development environment** option creates a branch named `vercel-dev` and sets Vercel development environment variables for it. The `vercel-dev` branch is a clone of your project's primary branch (`main`) that you can modify without affecting data on your primary branch.

>           The **Automatically delete obsolete Neon branches** option automatically deletes a Neon preview branch when the git branch that triggered its creation is merged or deleted.

Reviewer: ```suggestion
          With **Automatically delete obsolete Neon branches** enabled, Neon preview branches will be deleted whenever the git branch that triggered its creation is merged or deleted.
```

Reviewer: Slight chance of possible confusion in the original text, unless I'm misunderstanding, enabling this feature will delete branches going forward, not existing branches. Not 100% sure this revised text eliminates the possible confusion.
